src/Sylius/Bundle/CoreBundle/Resources/config/validation.xml

```diff
@@ -152,4 +152,13 @@
         </property>
     </class>
 
+    <class name="Sylius\Component\Core\Model\Shipment">
+        <property name="method">
+            <constraint name="NotBlank">
+                <option name="message">sylius.shipment.shipping_method.not_blank</option>
+                <option name="groups">sylius_checkout_select_shipping</option>
```

Comment:
The shipping method can't be null on shipment, we should use default `sylius` group and put this validation in `ShippingBundle`